EDKII-DRAFT-RFCs/edkii-crypto-layout.md

```diff
@@ -0,0 +1,561 @@
+# RFC: 0000: EDK2 Crypto Architecture Reorganization
+
+## 1. Summary
+
+This proposal reorganizes EDK2's cryptographic components to improve modularity, maintainability, and security response time. The core changes are:
+
+1. **CryptoPkg remains in EDK2** as the standardized cryptographic interface layer (BaseCryptLib.h, TlsLib.h, HashApiLib.h)
+2. **Crypto provider implementations move to edk2-crypto repository** - OpensslPkg and MbedTlsPkg are promoted to independent packages
+3. **BaseCryptLib.h becomes the enforced contract** that all crypto providers must implement
+
+---
+
+## 2. Motivation
+
+The current monolithic CryptoPkg architecture creates several challenges that this proposal aims to address:
+
+### Independent Security Response
+
+Critical OpenSSL/MbedTLS vulnerabilities currently require full EDK2 release cycles, delaying security patches by weeks or months. By separating crypto providers into independent packages, security fixes can be released immediately without waiting for EDK2 coordination.
```

Comment:
Consider rephrasing from "without waiting for EDK2 coordination". I'm sure there will still be some coordination, but there is less overhead as the crypto provider update is detached from immediately impacting more broad use cases in the edk2 repository.